info-compiler/src/main/java/com/google/sps/webcrawler/NewsContentProcessor.java

```diff
@@ -15,23 +15,236 @@
 package com.google.sps.webcrawler;
 
 import com.google.sps.data.NewsArticle;
+import com.google.sps.infocompiler.Config;
+import java.io.FileInputStream; 
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import opennlp.tools.sentdetect.SentenceDetectorME;
+import opennlp.tools.sentdetect.SentenceModel;
+import opennlp.tools.tokenize.TokenizerME;
+import opennlp.tools.tokenize.TokenizerModel;
+import org.apache.commons.math3.linear.ArrayRealVector;
+import org.apache.commons.math3.linear.RealVector;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.graph.Edge;
+import org.apache.flink.graph.Graph;
+import org.apache.flink.graph.Vertex;
+import org.apache.flink.graph.library.linkanalysis.PageRank;
+import org.apache.flink.types.DoubleValue;
+import org.apache.flink.types.IntValue;
+import org.apache.flink.types.StringValue;
 
 /** Static utilities for processing textual content, such as abbreviation. */
 public class NewsContentProcessor {
   static final int MAX_WORD_COUNT = 100;
+  private final static String SENTENCE_DETECTOR_FILE = Config.OpenNLP_SENTENCE_DETECTOR_FILE;
+  private final static String TOKENIZER_FILE = Config.OpenNLP_TOKENIZER_FILE;
+  private final static double PAGERANK_DAMPEN_FACTOR = 0.1;
+  private final static int PAGERANK_MAX_ITER = 100;
+  private final static double PAGERANK_CONVERGENCE_THRESHOLD = 0.0001;
+  private final static boolean PAGERANK_INCLUDE_ZERO_DEGREES_VERTICES = true;
+  private final static double SIMILARITY_THRESHOLD = 0.2;
+  private final static int SUMMARIZATION_MAX_SENTENCE_NUMBER = 3;
+  private static class PAGERANK_SCORE_DESCENDING implements Comparator<PageRank.Result> {
+    @Override
+    public int compare(PageRank.Result a, PageRank.Result b) {
+      // Higher score comes first. With the same score, smaller index (sentences that appear early)
+      // comes first.
+      if (a.getPageRankScore().getValue() < b.getPageRankScore().getValue() ||
+          (a.getPageRankScore().getValue() == b.getPageRankScore().getValue() && 
+              ((IntValue) a.getVertexId0()).getValue()
+                  > ((IntValue) b.getVertexId0()).getValue())) {
+        return 1;
+      } else {
+        return -1;
+      }
+    }
+  }
+  private static class SENTENCES_ORIGINAL_ORDER implements Comparator<PageRank.Result> {
```

Comment:
Don't caps class name: https://google.github.io/styleguide/javaguide.html#s5.2.2-class-names

Reply:
Same for PAGERANK_SCORE_DESCENDING

Reply:
Got it, thanks for pointing this out!
I referenced the walkthrough from a few weeks ago, and I think it might be better to directly instantiate a constant `Comparator` object, instead of creating a new object each time. Let me know if this is okay!